2016/proposals/narayan-aditya-visualizer.md

```diff
@@ -0,0 +1,194 @@
+   Returns the responses for the particular survey, listed
+   according to the questions contained in the survey.
+ * POST `/api/workshop/<workshop_id>/survey/<survey_id>/refetch`  
+   Instructs the Django application to fetch the responses
+   that the survey has received since the last time the
+   SurveyMonkey API was hit.
+
+The underlying Django app will perform these tasks:
+ * Polling SurveyMonkey API management command:
+   This management command would populate the database with survey
+   responses fetched from the [SurveyMonkey API].
+   The number of survey responses obtained from the [get_response_counts]
+   API call will be compared to the number of survey responses
+   saved in the database. The remaining responses will be fetched
+   using the [get_respondent_list] and [get_responses] API calls.
+ * Segregating the survey responses according to the answer to the
+   question: *Which workshop did you attend?* or the custom variable
+   `workshop_id`.
+ * Rendering and serving the templates.
+
+### Scheduler - Cron
+
+[Cron] will be used as a job scheduler to periodically run
+the *Polling SM API* management command at fixed time intervals.
+The interval duration would be dynamically determined by the
+number of API calls left and the time of the day.
+
+### Database - PostgreSQL
+
+The Django app will use [PostgreSQL] as its database engine to
+store the survey responses fetched by the backend.
+
+### Frontend - Google Charts and Bootstrap
+
+The user interface would use charts and graphs to visualize the
+large amount of data aggregated from the survey responses.
+[Google Charts] will be used to display interactive charts on
+the user interface while it communicates with the API defined
+before.
+
+The surveys contain a wide range of questions with multiple and
+single option choices. Google Charts provides us with a very large
+collection of chart types to experiment with.
+The ones that would be most effective in conveying the information
+include:
+ * Pie Charts: Pie Charts can be used for questions with limited
+   number of answer choices (usually binary).
+ * Bar Charts: Bar Charts will be used to compare numeric value of
+   a wide range of questions against one another.
+ * Table: Text inputs are tough to visualize and tables provide us
+   an easy way to list them.
+
+A [sample application] involving the use of Bar Charts to compare
+the number of responses for the various answers to the questions
+listed in the [sample survey] can be viewed [here].
+
+## Schedule of Deliverables
+
+### Community Bonding Period
+During this period, the finer technical details and decisions will
+be taken in consultation with the mentors. These include:
+ * The charts to be used to visualize each question in the survey.
+ * The database schema that would allow aggregation and segregation
+   of survey responses according to workshops.
+
+### May 25th -  June 7th
+Start implementation of the *Polling SM API* management command
```

Comment:
I have already managed to poll the SM API to get survey responses in the sample Django application. See [here](https://github.com/narayanaditya95/survey-visualiser/blob/master/visualiser/views.py#L14). It would be a slight upgrade from the existing code.
Are there concerns regarding working on this in the beginning?